src/ucp/core/ucp_rkey.c

```diff
@@ -213,61 +222,321 @@ UCS_PROFILE_FUNC(ssize_t, ucp_rkey_pack_memh,
                                 sys_dev_map, sys_distance, buffer, 1, 0);
 }
 
+static size_t ucp_memh_common_packed_size()
+{
+    size_t size;
+
+    /* Size of mkey information which comes prior TL mkey data of all MDs */
+    size = sizeof(uint16_t);
+
+    /* Flags */
+    size += sizeof(uint64_t);
+
```

Comment:
is this relevant only for packing exported keys?
 Why is then `common` in the name?
Why do we need to pack `size`?
Do we really need 8 bytes for flags?

Reply:
> Do we really need 8 bytes for flags

reserved for the future

> Why do we need to pack size?

yes, to support extending new fields in the future

> Why is then common in the name?

will be used for exported and RKEY new formats.

> is this relevant only for packing exported keys?

for RKEYv2 and exported